Perf run for #72412

tmandry opened the pull request:
Just creating this to run perf for #72412 without the last commit.

cc @VFLashM 

r? @ghost

tmandry commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion

bors commented:
:hourglass: Trying commit 6ebb3050634c5973ccc1883fcd05e41667a892e6 with merge 9dbf9cc98c62a31b10d247f963be718bcc4e358e...
<!-- homu: {"type":"TryBuildStarted","head_sha":"6ebb3050634c5973ccc1883fcd05e41667a892e6","merge_sha":"9dbf9cc98c62a31b10d247f963be718bcc4e358e"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/1131715456), [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=36675)
Build commit: 9dbf9cc98c62a31b10d247f963be718bcc4e358e (`9dbf9cc98c62a31b10d247f963be718bcc4e358e`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-azure":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=36675","checks-actions":"https://github.com/rust-lang-ci/rust/runs/1131715456"},"merge_sha":"9dbf9cc98c62a31b10d247f963be718bcc4e358e"} -->

rust-timer commented:
Queued 9dbf9cc98c62a31b10d247f963be718bcc4e358e with parent f3c923a13a458c35ee26b3513533fce8a15c9c05, future [comparison URL](https://perf.rust-lang.org/compare.html?start=f3c923a13a458c35ee26b3513533fce8a15c9c05&end=9dbf9cc98c62a31b10d247f963be718bcc4e358e).

rust-timer commented:
Finished benchmarking try commit (9dbf9cc98c62a31b10d247f963be718bcc4e358e): [comparison url](https://perf.rust-lang.org/compare.html?start=f3c923a13a458c35ee26b3513533fce8a15c9c05&end=9dbf9cc98c62a31b10d247f963be718bcc4e358e).

Benchmarking this pull request likely means that it is perf-sensitive, so we're automatically marking it as not fit for rolling up. Please note that if the perf results are neutral, you should likely undo the rollup=never given below by specifying `rollup-` to bors.

Importantly, though, if the results of this run are non-neutral **do not** roll this PR up -- it will mask other regressions or improvements in the roll up.

@bors rollup=never

tmandry commented:
Moved to the latest commit from #72412 and removed the extra Ty in the key just to see if that helps. Not sure it will matter too much for instruction counts though. Maybe we should tweak the SSO size a bit?

@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion

bors commented:
:hourglass: Trying commit 4f7233f9f64f1c3b21b5181e0b3a0edbed9d681b with merge 7623109879ed842c783c7726803439b8f8c185d0...
<!-- homu: {"type":"TryBuildStarted","head_sha":"4f7233f9f64f1c3b21b5181e0b3a0edbed9d681b","merge_sha":"7623109879ed842c783c7726803439b8f8c185d0"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/1132588610), [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=36680)
Build commit: 7623109879ed842c783c7726803439b8f8c185d0 (`7623109879ed842c783c7726803439b8f8c185d0`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-azure":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=36680","checks-actions":"https://github.com/rust-lang-ci/rust/runs/1132588610"},"merge_sha":"7623109879ed842c783c7726803439b8f8c185d0"} -->

rust-timer commented:
Queued 7623109879ed842c783c7726803439b8f8c185d0 with parent f3c923a13a458c35ee26b3513533fce8a15c9c05, future [comparison URL](https://perf.rust-lang.org/compare.html?start=f3c923a13a458c35ee26b3513533fce8a15c9c05&end=7623109879ed842c783c7726803439b8f8c185d0).

rust-timer commented:
Finished benchmarking try commit (7623109879ed842c783c7726803439b8f8c185d0): [comparison url](https://perf.rust-lang.org/compare.html?start=f3c923a13a458c35ee26b3513533fce8a15c9c05&end=7623109879ed842c783c7726803439b8f8c185d0).

Benchmarking this pull request likely means that it is perf-sensitive, so we're automatically marking it as not fit for rolling up. Please note that if the perf results are neutral, you should likely undo the rollup=never given below by specifying `rollup-` to bors.

Importantly, though, if the results of this run are non-neutral **do not** roll this PR up -- it will mask other regressions or improvements in the roll up.

@bors rollup=never

VFLashM commented:
> Maybe we should tweak the SSO size a bit?

My benchmarks show that `MiniSet` is faster than `FxHashSet` for up to 64 elements of small storage.
I originally had it at `[T, 64]`.

However, in practice, in vast, vast, vast majority of cases amount of elements in it was less than 3.
After seeing that I shrunk it to 8 in attempt to be more cache-friend and achieve better locality.
